### DISCUSSION TITLE
Adding newly created node to SelectionService

### DISCUSSION THREAD

**Discussion by we-wake-act21** [eclipse-glsp/glsp#1038] [Q&A] [answered]
Hi,

while adding new node to existing GLSP Diagram, I want this new node to be shown as selected. How can I achieve this??

I tried to call the SelectAction after the node is added to the GModel `this.actionDispatcher.dispatch(new SelectAction(Arrays.asList(id)));` but this has no effect. 

Is there any other way?

**Comment by martin-fleck-at** [comment, marked as answer]
@we-wake-act21 I believe the problem is that after you create the node, the GModel will be re-created and updated and only afterwards you can actually do the select. On the server side we have an implementation of a queue where you can [add actions that are applied after the next update](https://github.com/eclipse-glsp/glsp-server/blob/fffccca5d367dcfd5555349307dc6800b4ce422e/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/internal/actions/DefaultActionDispatcher.java#L109). It is used, for instance, in the [GModelCreateNodeOperationHandler](https://github.com/eclipse-glsp/glsp-server/blob/fffccca5d367dcfd5555349307dc6800b4ce422e/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/gmodel/GModelCreateNodeOperationHandler.java#L62). I believe you need to do something similar.
